src/CoreBundle/private/library/classes/TCMSURLHistory.class.php

```diff
@@ -78,26 +85,35 @@ public function getSimilarHistoryElementIndex(array $newElementParameters): ?int
      */
     public function GetURL()
     {
-        if ($this->index > 0) {
-            $url = $this->aHistory[($this->index - 1)]['url'];
+        $historyIndex = $this->getHistoryIndex();
 
-            return $url;
+        if ($historyIndex > 0 && isset($this->aHistory[$historyIndex - 1])) {
+            return $this->aHistory[$historyIndex - 1]['url'];
         } else {
             return false;
         }
     }
 
+    public function getHistoryIndex()
+    {
+        return count($this->aHistory);
+    }
+
     /**
      * returns and removes the url of the last history element.
      *
      * @return bool|string
      */
     public function PopURL()
     {
-        if ($this->index > 0) {
+        $historyIndex = $this->getHistoryIndex();
+
+        if ($historyIndex > 0) {
             $url = $this->GetURL();
-            unset($this->aHistory[($this->index - 1)]);
-            --$this->index;
+            unset($this->aHistory[$historyIndex - 1]);
+
+            // reset the keys to prevent key gaps
+            $this->aHistory = array_values($this->aHistory);
```

Comment:
Is this really necessary? We remove the last element after all, so no indices should change.

I think we can replace the complete method body with a call to `array_pop` (we only need to return false if array_pop returns null).

Reply:
done

Reply:
Almost - now we return the whole history entry, not just the URL part as before.

Reply:
gnarf... fixed it